tests/validators/answers/test_answer_validator.py

```diff
@@ -33,6 +41,108 @@ def test_invalid_single_date_period():
         "type": "Date",
     }
 
-    answer_validator = DateAnswerValidator(answer)
+    answer_validator = DateAnswerValidator(
+        answer, get_mock_schema_with_data_version("0.0.3")
+    )
 
     assert not answer_validator.is_offset_date_valid()
+
+
+def test_confirmation_question_q_code():
+    filename = "schemas/valid/test_q_codes.json"
+    schema = QuestionnaireSchema(_open_and_load_schema_file(filename))
+    answer = {
+        "type": "Radio",
+        "id": "confirmation-1-answer",
+        "q_code": "1",
+        "options": [
+            {"label": "Yes this is correct", "value": "Yes this is correct"},
+            {"label": "No I need to change this", "value": "No I need to change this"},
+        ],
+        "mandatory": True,
+    }
```

Comment:
This could be simplified.
```suggestion
    answer = schema.get_answer("confirmation-1-answer")
    answer["q_code"] = "1"
```